### PR TITLE
Fix preserving message processor state

### PR DESCRIPTION
**Pull request by bsenduran**
By introducing a state in registry, the message processor's state is perserved.
Issue: wso2/product-ei#849

### CODE REVIEW

File: modules/core/src/main/java/org/apache/synapse/message/processor/impl/ScheduledMessageProcessor.java [hunk 1]
```diff
@@ -81,6 +83,12 @@
 
 	private int memberCount = 1;
 
+	private Registry registry;
+
+	private static final String REG_PROCESSOR_BASE_PATH = "/repository/components/org.apache.synapse.message.processor/";
+
+	private static final String MP_STATE = "MESSAGE_PROCESSOR_STATE";
```
Comment (milindaperera) [inline modules/core/src/main/java/org/apache/synapse/message/processor/impl/ScheduledMessageProcessor.java]: Shall we add little description about this property here?

Comment (bsenduran) [inline modules/core/src/main/java/org/apache/synapse/message/processor/impl/ScheduledMessageProcessor.java]: isn't this property self descriptive enough

Comment (milindaperera) [inline modules/core/src/main/java/org/apache/synapse/message/processor/impl/ScheduledMessageProcessor.java]: 👍

Comment (milindaperera) [inline modules/core/src/main/java/org/apache/synapse/message/processor/impl/ScheduledMessageProcessor.java]: Any way, I felt better if we have at least what are the processor states without looking ussage of this property. That's why I suggested.

**Pull request merged**